enos/enos-scenario-upgrade.hcl

```diff
@@ -266,25 +303,104 @@ EOF
     ]
   }
 
-  step "upgrade_clients" {
+  step "upgrade_first_client" {
```

Comment:
We're now effectively hard-coding the number of clients we run, because we can't use a `for_each` here (that'd be in parallel instead of serially). That's probably fine for the MVP but we might want to think about if there's a way to make that more flexible in the future.